docs/design/architecture.md

```diff
@@ -2,128 +2,353 @@
 
 ## Guest assets
 
-The hypervisor will launch a virtual machine which includes a minimal guest kernel
-and a guest image.
+To create a VM to host a container, Kata Containers launches a
+hypervisor. The hypervisor needs two assets to create a VM: a kernel
+and an image file.
 
 ### Guest kernel
 
-The guest kernel is passed to the hypervisor and used to boot the virtual
-machine. The default kernel provided in Kata Containers is highly optimized for
-kernel boot time and minimal memory footprint, providing only those services
-required by a container workload. This is based on a very current upstream Linux
-kernel.
+The Linux [guest kernel](../../tools/packaging/kernel)
+is passed to the hypervisor and used to boot the VM.
+The default kernel provided in Kata Containers is highly optimized for
+kernel boot time and minimal memory footprint, providing only those
+services required by a container workload. This is based on a very
+current upstream Linux kernel.
 
 ### Guest image
 
-Kata Containers supports both an `initrd` and `rootfs` based minimal guest image.
+The hypervisor uses an image file which provides a minimal root
+filesystem used by the guest kernel to boot the VM and host the Kata Container.
+
+Kata Containers supports both initrd and rootfs based minimal guest images.
+
+The default images are created using the [`osbuilder`](../../tools/osbuilder) tool.
+
+> **Note:**
+>
+> The guest image is *unrelated* to the image used in a container
+> workload.
+>
+> For example, if a user creates a container that runs a shell in a
+> BusyBox image, they will run that shell in a BusyBox environment.
+> However, the guest image running inside the VM that is used to
+> *host* that BusyBox image could be running Clear Linux, Ubuntu,
+> Fedora or any other distribution potentially.
+>
+> The `osbuilder` tool provides
+> [configurations for various common Linux distributions](../../tools/osbuilder/rootfs-builder)
+> which can be built into either initrd or rootfs guest images.
 
 #### Root filesystem image
 
-The default packaged root filesystem image, sometimes referred to as the "mini O/S", is a
-highly optimized container bootstrap system based on [Clear Linux](https://clearlinux.org/). It provides an extremely minimal environment and
-has a highly optimized boot path.
+The default packaged root filesystem image ("rootfs"), sometimes
+referred to as the "mini O/S", is a highly optimized container
+bootstrap system based on [Clear Linux](https://clearlinux.org). It
+provides an extremely minimal environment and has a highly optimized
+boot path.
+
+The only services running in the context of the mini O/S are:
+
+| Process | Environment | systemd service? | Notes |
+|-|-|-|-|
+| systemd | VM root | n/a | The init daemon, running as PID 1 |
+| [Agent](#agent) | VM root | yes | Runs as a systemd service |
+| `chronyd` | VM root | yes | Used to synchronise the time with the host |
+| container workload | VM container | no | Managed by the agent |
```

Comment:
I would consider another column   "User accessible"  The Agent isn't directly accessible by the user but the container workload is.